awx/main/tasks/system.py

```diff
@@ -454,6 +451,10 @@ def inspect_execution_nodes(instance_list):
             instance.last_seen = last_seen
             instance.save(update_fields=['last_seen'])
 
+            # Make sure that hop nodes don't fall through and have the execution_node_health_check task applied
+            if not any(cmd['WorkType'] == 'ansible-runner' for cmd in ad['WorkCommands'] or []):
+                continue
```

Comment:
yeah, this is probably not needed anymore, but it looks good as a defensive coding measure.

Reply:
Marking this for posterity: we decided on a clearer, more-readable pattern for these checks.